dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunctionLogMessages.cs

```diff
@@ -36,13 +38,47 @@ public static partial void LogFunctionInvoking(
             logLevel: LogLevel.Trace,   // Sensitive data, logging as trace, disabled by default
             eventId: 0,
             "Function arguments: {Arguments}");
+
+    [RequiresUnreferencedCode("Uses reflection to serialize function arguments, making it incompatible with AOT scenarios.")]
+    [RequiresDynamicCode("Uses reflection to serialize the function arguments, making it incompatible with AOT scenarios.")]
     public static void LogFunctionArguments(this ILogger logger, KernelArguments arguments)
+    {
+        LogFunctionArgumentsInternal(logger, arguments);
+    }
+
+    /// <summary>
+    /// Logs arguments of a <see cref="KernelFunction"/>.
+    /// </summary>
+    public static void LogFunctionArguments(this ILogger logger, KernelArguments arguments, JsonSerializerOptions jsonSerializerOptions)
+    {
+        LogFunctionArgumentsInternal(logger, arguments, jsonSerializerOptions);
+    }
+
+    /// <summary>
+    /// Logs arguments of a <see cref="KernelFunction"/>.
+    /// </summary>
+    private static void LogFunctionArgumentsInternal(this ILogger logger, KernelArguments arguments, JsonSerializerOptions? jsonSerializerOptions = null)
     {
         if (logger.IsEnabled(LogLevel.Trace))
         {
             try
             {
-                var jsonString = JsonSerializer.Serialize(arguments);
+                string jsonString;
+
+                if (jsonSerializerOptions is not null)
+                {
+                    JsonTypeInfo<KernelArguments> typeInfo = (JsonTypeInfo<KernelArguments>)jsonSerializerOptions.GetTypeInfo(typeof(KernelArguments));
+                    jsonString = JsonSerializer.Serialize(arguments, typeInfo);
+                }
+                else
+                {
+#pragma warning disable IL2026 // Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code
```

Comment:
Pragma suppressions won't prevent the linker from issuing warnings when users publish their final trimmed/AOT application. You should instead use `UnconditionalSuppressAttribute` annotations on the method or type itself, for example:

https://github.com/dotnet/runtime/blob/04c5a32aa4dc6b2f07440debd221f00d8a1582b1/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectConverterFactory.cs#L37-L41

Even then though, the attribute should only be applied in methods that are justifiable only invoked in unsafe code paths. In the linked example we can suppress unsafe instance methods because all constructors have been marked unsafe.

Reply:
Thanks for pointing out, I did not know that.